syntax = "proto3";

package mypackage;

message CreateFileRequest {
  string file_name = 1;
  bytes file_contents = 2;
}

message CreateFileResponse {
  bool success = 1;
}

message DeleteFileRequest {
  string file_name = 1;
}

message DeleteFileResponse {
  bool success = 1;
}

message WriteFileRequest {
  string file_name = 1;
  bytes data = 2;
}

message WriteFileResponse {
  bool success = 1;
}

message ReadFileRequest {
  string file_path = 1;
}

message ReadFileResponse {
  bool success = 1;
  string file_contents = 2;
  string error_message = 3;
}

service MyService {
  rpc CreateFile(CreateFileRequest) returns (CreateFileResponse);
  rpc DeleteFile(DeleteFileRequest) returns (DeleteFileResponse);
  rpc WriteFile(WriteFileRequest) returns (WriteFileResponse);
  rpc ReadFile(ReadFileRequest) returns (ReadFileResponse);
}